projects/cps-ui-kit/src/lib/components/cps-autocomplete/cps-autocomplete.component.ts

```diff
@@ -664,14 +670,22 @@ export class CpsAutocompleteComponent
     );
   }
 
-  onBeforeOptionsHidden(reason: CpsMenuHideReason) {
+  onBeforeOptionsHidden(reason: CpsMenuHideReason): boolean | void {
+    if (
+      (this.loading || this.validating) &&
+      reason !== CpsMenuHideReason.KEYDOWN_ESCAPE
+    ) {
+      // Prevent hiding the options menu when loading or validating
+      return false;
+    }
```

Comment:
Does the returned value (`true` or `false`) change the behavior?

Reply:
@lukasmatta good catch! fixed